sources/packages/backend/apps/queue-consumers/src/processors/virus-scan/_tests_/virus-scan.processor.e2e-spec.ts

```diff
@@ -40,6 +40,158 @@ describe(describeProcessorRootTest(QueueNames.FileVirusScanProcessor), () => {
     jest.resetAllMocks();
   });
 
+  it("Should throw an error when the student file is not found during virus scanning.", async () => {
+    // Arrange
+    const studentFile = createFakeStudentFileUpload();
+    studentFile.virusScanStatus = VirusScanStatus.InProgress;
+    await db.studentFile.save(studentFile);
+    const fakeUniqueFileName = "file.random";
+    const mockedJob = mockBullJob<VirusScanQueueInDTO>({
+      uniqueFileName: fakeUniqueFileName,
+      fileName: studentFile.fileName,
+    });
+
+    // Act
+    const errorMessage = `File ${fakeUniqueFileName} is not found or has already been scanned for viruses. Scanning the file for viruses is aborted.`;
+    await expect(
+      processor.performVirusScan(mockedJob.job),
+    ).rejects.toStrictEqual(new Error(errorMessage));
+    expect(
+      mockedJob.containLogMessages([
+        "Log details",
+        "Starting virus scan.",
+        errorMessage,
+      ]),
+    ).toBe(true);
+  });
+
+  it("Should throw an error when the connection to the virus scan server failed.", async () => {
+    // Arrange
+    const studentFile = createFakeStudentFileUpload();
+    studentFile.virusScanStatus = VirusScanStatus.InProgress;
+    await db.studentFile.save(studentFile);
+    const errorMessage = `Unable to scan the file ${studentFile.uniqueFileName} for viruses. Connection to ClamAV server failed.`;
+    clamAVServiceMock.scanFile = jest.fn(() => {
+      throw new ClamAVError(errorMessage, "ECONNREFUSED");
```

Comment:
The error message here can be misleading since it is from the lib not from SIMS.
A generic dummy message would be preferred.